core/unix/coredump.c

```diff
@@ -484,11 +484,17 @@ os_dump_core_internal(dcontext_t *dcontext)
     // of LOAD program header (number of sections minus one since shstrtab does not
```

Comment:
> i#7046: Add padding to align section data to page size. #7144

The title is misleading: this is not about section data; this is about the section header table.

`s/section data/section header table/`

Reply:
I've updated the title to clarify it's the loadable segment.

Reply:
Here you again have "loadable segment": then I'm very confused: could you clarify?  Isn't this affecting the section headers?  The thing you're changing is the parameter "section_header_table_offset" to write_elf_header.

Reply:
Here's the layout of the core file:

ELF header
Program headers
content (loadable segments in our case)
Section headers

The padding is added between program headers and content/loadable segments so that loadable segments are page size aligned. For example, offset 0x1000 in the example instead of 0xe00 without the padding.

With the padding, the section headers are moved, increasing the value of "Start of section headers" which is "section_header_table_offset". Adding the padding before the content fixes the alignment. The change in section_header_table_offset is the side effect of the padding.

Reply:
OK, this is much clearer now.